### ISSUE TITLE
ERR_TUNNEL_CONNECTION_FAILED with forwardproxy's naive

### ISSUE THREAD

**Issue by biaocy** [klzgrad/naiveproxy#91]
Hi, first of all, thank you for creating this wonderful project.

I have built Caddy v2 with naive fork of forwardproxy, and setup like [README's setup](https://github.com/klzgrad/naiveproxy#setup), and built naive from the master branch as client side.

My environment:
- OS:
  - Server: Ubuntu 20.04
  - Local: Ubuntu 18.04
- Caddy config: Just like the example from [README's setup](https://github.com/klzgrad/naiveproxy#setup) replaced some example values
- Naive client config: 
```json
{
  "listen": "socks://127.0.0.1:1080",
  "proxy": "https://user:pass@subdomain.example.com",
  "log": ""
}
```

I tested with this command `curl -x 'socks5h://127.0.0.1:1080' https://google.com`, and it return this error:
```shell
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to google.com:443 
```
and naive client logs:
```shell
[0617/225433.315632:INFO:naive_proxy_bin.cc(418)] Proxying via https://***.***.com
[0617/225433.315994:INFO:naive_proxy_bin.cc(552)] Listening on 127.0.0.1:1080
[0617/225612.100683:INFO:naive_connection.cc(250)] Connection 1 to 216.58.200.238:443
[0617/225613.606668:INFO:naive_proxy_delegate.cc(82)] Padding capability of https://***.***.com:443 undetected
[0617/225613.606972:INFO:naive_proxy.cc(178)] Connection 1 closed: ERR_TUNNEL_CONNECTION_FAILED
[0617/225625.828105:INFO:naive_connection.cc(250)] Connection 2 to 104.28.18.94:443
[0617/225625.924111:INFO:naive_proxy.cc(178)] Connection 2 closed: OK
[0617/225634.910677:INFO:naive_connection.cc(250)] Connection 3 to 69.171.229.28:443
[0617/225635.589165:INFO:naive_proxy.cc(178)] Connection 3 closed: OK
[0617/225650.755351:INFO:naive_connection.cc(250)] Connection 4 to 69.171.229.28:443
[0617/225651.290740:INFO:naive_proxy.cc(178)] Connection 4 closed: OK
[0617/225659.015342:INFO:naive_connection.cc(250)] Connection 5 to 104.244.42.65:443
[0617/225659.113133:INFO:naive_proxy.cc(178)] Connection 5 closed: OK
[0617/225707.591545:INFO:naive_connection.cc(250)] Connection 6 to 69.171.244.15:443
[0617/225707.712276:INFO:naive_proxy.cc(178)] Connection 6 closed: OK
```

After first request failed, every subsequent request immediate returns `Connection closed: OK` as naive's log said. And all `curl` command return `curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL`.

I'm also tested with `chromium 83.0.4103.61` to visit `https://google.com`, and it returns `www.google.com unexpectedly closed the connection.`

Is it possible that these requests has been detected?

**Comment by klzgrad** [comment]
```
Padding capability of https://***.***.com:443 undetected
```
If you Caddy server is set up correctly this line shouldn't appear. Probably you had the wrong passwords or something else that didn't route proxy requests to forwardproxy.

**Comment by biaocy** [comment]
Something I'm forgot to mentioned, I directly visit the server domain in browser, it returns index.html correctly. 

This time, I changed my user/pass to a simple one, and double checked. Here is the new naive's log:
```shell
[0618/202929.872652:INFO:naive_proxy_bin.cc(418)] Proxying via https://***.***.com
[0618/202929.872849:INFO:naive_proxy_bin.cc(552)] Listening on 127.0.0.1:2080
[0618/203009.839259:INFO:naive_connection.cc(250)] Connection 1 to google.com:443
[0618/203409.941359:INFO:naive_proxy.cc(178)] Connection 1 closed: ERR_PROXY_CONNECTION_FAILED
[0618/203422.458787:INFO:naive_connection.cc(250)] Connection 2 to www.google.com:443
[0618/203422.660424:INFO:naive_proxy_delegate.cc(82)] Padding capability of https://***.***.com:443 undetected
[0618/203422.660691:INFO:naive_proxy.cc(178)] Connection 2 closed: ERR_TUNNEL_CONNECTION_FAILED
[0618/203435.639018:INFO:naive_connection.cc(250)] Connection 3 to twiiter.com:443
[0618/203435.664491:INFO:naive_proxy.cc(178)] Connection 3 closed: OK
[0618/203444.040476:INFO:naive_connection.cc(250)] Connection 4 to youtube.com:443
[0618/203444.066754:INFO:naive_proxy.cc(178)] Connection 4 closed: OK
```
And the curl command returns the same error as last time requested.

**Comment by klzgrad** [comment]
This is something only you can reproduce. I wouldn't know every detail of your environment. There are generally two ways of debugging the client side:
- With `./naive --ssl-key-log-file=/tmp/keys` and use `/tmp/keys` in Wireshark https://wiki.wireshark.org/TLS#TLS_Decryption to see the actual requests.
- With `./naive --log-net-log=/tmp/log.json`, and use https://netlog-viewer.appspot.com/ to inspect the NetLog.

Don't post these logs publicly though. Also, you should check Caddy's log and see if forwardproxy is actually functioning.

**Comment by biaocy** [comment]
Sorry for the delayed response and my poor knowledge in networking.
Thank you so much for pointing out the direction I can investigate.

**TLDR: This issue is caused by cloudflare's proxy. It will work after cloudflare's proxy turn off**

Today, I inspected the NetLog, although I have no knowledge about this. Nevertheless, I try to find some useful information out of it.
And I saw something about `cloudflare` shown up. So I wonder if this issue is caused by cloudflare proxy.

After I turn off cloudflare proxy, it work like a charm.
Thank you for creating such a wonderful project.
